functions/src/notifications.ts

```diff
@@ -0,0 +1,90 @@
+import { firestore, EventContext } from 'firebase-functions'
+import * as admin from 'firebase-admin'
+import * as constants from './constants'
+import * as algoliasearch from 'algoliasearch'
+try { admin.initializeApp() } catch (e) {}
+
+type ActivityData = {
+    activityId: string,
+    ownerId: string,
+    topicIds: string[],
+    coords: FirebaseFirestore.GeoPoint,
+}
+
+export const onActivityCreation = firestore.document('activities/{activityId}').onCreate(activityCreationHandler)
+
+export async function activityCreationHandler(snap: FirebaseFirestore.DocumentSnapshot, context: EventContext) {
+    const newData = exports.getActivityData(snap)
```

Comment:
Here, we call `exports` so that it's the same thing that we're spying on in tests

Reply:
Ahhh! This is the thing I didn't do!  How'd you figure that out?

Mocking is sooo hard when you don't have a great grasp of the module management of a language.